server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -990,12 +1261,48 @@ public void cleanFiles(int totalTranslogOps, long globalCheckpoint, Store.Metada
                                ActionListener<Void> listener) {
         }
 
+        @Override
+        public void downloadSnapshotFile(String repository, IndexId indexId, BlobStoreIndexShardSnapshot.FileInfo snapshotFile,
```

Comment:
Seems odd to have all these methods that just throw away their listeners, should we be asserting that they're not called? Not really important in this PR tho.